packages/core/src/state/NetworkManager.ts

```diff
@@ -123,14 +143,20 @@ export default class NetworkManager implements Manager {
           );
           throw error;
         });
-    let promise;
+      // legacy behavior schedules resolution after dispatch
+      if (!throttle && !action.endpoint) {
+        promise = resolvePromise(promise);
+      }
+      return promise;
+    };
+
     if (throttle) {
-      promise = this.throttle(key, deferedFetch);
+      return this.throttle(key, deferedFetch)
+        .then(data => resolve(data))
+        .catch(error => reject(error));
     } else {
-      promise = deferedFetch();
+      return deferedFetch().catch(() => {});
```

Comment:
```suggestion
      return deferedFetch().catch();
```

Reply:
Removing the function makes the exception fallthrough and the tests fail.